x-pack/plugin/mapper-unsigned-long/src/main/java/org/elasticsearch/xpack/unsignedlong/UnsignedLongField.java

```diff
@@ -0,0 +1,74 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+package org.elasticsearch.xpack.unsignedlong;
+
+import org.elasticsearch.script.Converter;
+import org.elasticsearch.script.Converters;
+import org.elasticsearch.script.Field;
+import org.elasticsearch.script.FieldValues;
+
+import java.math.BigInteger;
+import java.util.List;
+import java.util.stream.Collectors;
+
+import static org.elasticsearch.xpack.unsignedlong.UnsignedLongFieldMapper.BIGINTEGER_2_64_MINUS_ONE;
+
+public class UnsignedLongField extends Field.LongField {
+    public UnsignedLongField(String name, FieldValues<Long> values) {
+        super(name, values);
+    }
+
+    // UnsignedLongFields must define their own conversions as they are in x-pack
+    @Override
+    public <CT, CF extends Field<CT>> Field<CT> as(Converter<CT, CF> converter) {
+        if (converter.getFieldClass().isInstance(this)) {
+            return converter.getFieldClass().cast(this);
+        }
+
+        if (converter.getTargetClass() == BigInteger.class) {
+            BigIntegerField bigIntegerField = UnsignedLongToBigInteger(this);
+            return converter.getFieldClass().cast(bigIntegerField);
+        }
+
+        return super.as(converter);
```

Comment:
`UnsignedLongScriptDocValues`/`UnsignedLongField` returns `long`s with exactly this property, values may be negative. 

I expect diligent users will perform "confirming" type conversions and never to take the original type from `field(String)`, in this case the a user who wants to handle possible negative values themselves would only be able to do a `field('ul').as(Field.Long)` confirming type conversion.

If we want to allow `field('ul').as(Field.UnsignedLong)`, we'll need to implement Field augmentation for painless.

Reply:
I understand the complication with Long type since that is the underlying native type returned here, but I don’t think conversion to string will work correctly? It should not show negative values, yet I think it will since it would just stringify the raw long value. A minimal test for this conversion would be good.

Reply:
`UnsignedLongScriptDocValues`/`UnsignedLongField` returns `long`s with exactly this property, values may be negative. 

I expect diligent users to always perform "confirming" type conversions and never to take the original type from `field(String)`, in this case the a user who wants to handle possible negative values themselves would only be able to do a `field('ul').as(Field.Long)` confirming type conversion.

Reply:
Added minimal test.

Reply:
Can you point at the test? I was specifically talking about conversion to string, to ensure that unsigned long values do not show up as negative values with that conversion.

Reply:
This PR does not implement conversion _to_ String fields, only _from_ String fields.  To BigInteger and to Long are implemented here.